### PR TITLE
Additional cipher_info getters

### PR DESCRIPTION
**Pull request by mfil**
## Description
While trying to add mbedtls 3 support to OpenVPN, we found that we need some additional getter functions that are not yet in the dev branch. This pull request adds functions that get the IV and block size from a `cipher_info` struct and functions that check if the IV or key size of a cipher are variable.

We also need some more getter functions in other parts of the library, but I figured it's easier if I keep the pull requests small.

## Status
**READY**

## Requires Backporting
NO  

## Migrations
NO

## Todos
- [x] Tests
- [x] Documentation
- [x] Changelog updated

### CODE REVIEW

File: include/mbedtls/cipher.h [hunk 1]
```diff
@@ -507,6 +507,77 @@ static inline const char *mbedtls_cipher_info_get_name(
...
+/**
+ * \brief       This function returns the size of the IV or nonce
+ *              for the cipher info structure, in bytes.
+ *
+ * \param info  The cipher info structure. This may be \c NULL.
+ *
+ * \return      The recommended IV size.
+ * \return      \c 0 for ciphers not using an IV or a nonce.
+ */
+static inline int mbedtls_cipher_info_get_iv_size(
```
Comment (gilles-peskine-arm) [inline include/mbedtls/cipher.h:519]: Why `int`? Surely an IV is an unsigned value that fits in memory, so this should be `size_t`.

The library does use `int` for unsigned values in some places, but for the last few years we've avoided that in new code unless it was absolutely necessary for backward compatibility.

Same with `get_block_size`.

File: include/mbedtls/cipher.h [hunk 2]
```diff
@@ -507,6 +507,77 @@ static inline const char *mbedtls_cipher_info_get_name(
...
+    return( (int) info->MBEDTLS_PRIVATE(iv_size) );
+}
+
+/**
+ * \brief        This function returns the block size of the given
+ *               cipher info structure.
+ *
+ * \param info   The cipher info structure. This may be \c NULL.
+ *
+ * \return       The block size of the cipher.
```
Comment (gilles-peskine-arm) [inline include/mbedtls/cipher.h:534]: Please document the unit (both bits and bytes are plausible).

Please document the value for ciphers that aren't block ciphers.

**Review by gilles-peskine-arm — CHANGES_REQUESTED** [review]
Looks mostly good to me (and thanks for including tests!), except that I think some types are wrong and some documentation is missing.

File: include/mbedtls/cipher.h (hunk 1 above)
Comment (mfil) [reply include/mbedtls/cipher.h]: I used `int` to match the corresponding `mbedtls_cipher_get_*` functions. I agree that `size_t` makes more sense.

File: include/mbedtls/cipher.h (hunk 2 above)
Comment (mfil) [reply include/mbedtls/cipher.h:534]: I updated the documentation for both `mbedtls_cipher_info_get_block_size` and `mbedtls_cipher_get_block_size`.

**Review by gilles-peskine-arm — APPROVED** [review]

**Review by davidhorstmann-arm — CHANGES_REQUESTED** [review]
A couple of nits, but looks good otherwise.

File: include/mbedtls/cipher.h [hunk 3]
```diff
@@ -507,6 +507,79 @@ static inline const char *mbedtls_cipher_info_get_name(
...
 }
 
+/**
+ * \brief       This function returns the size of the IV or nonce
+ *              for the cipher info structure, in bytes.
+ *
+ * \param info  The cipher info structure. This may be \c NULL.
+ *
+ * \return      The recommended IV size.
+ * \return      \c 0 for ciphers not using an IV or a nonce.
```
[the remark this answers is not in the document]
Comment (mfil) [reply include/mbedtls/cipher.h:517]: Fixed!

For future reference, do you prefer small stuff like that in separate commits, or should I squash it into the main commits and `push --force`?

File: tests/suites/test_suite_cipher.function [hunk 4]
```diff
@@ -19,6 +19,7 @@ static int check_cipher_info( mbedtls_cipher_type_t type,
                               const mbedtls_cipher_info_t *info )
 {
     size_t key_bitlen;
+    int block_size, iv_size;
```
[the remark this answers is not in the document]
Comment (mfil) [reply tests/suites/test_suite_cipher.function]: Fixed

File: include/mbedtls/cipher.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (gilles-peskine-arm) [reply include/mbedtls/cipher.h:517]: We prefer additional commits unless there's a good reason to rebase (e.g. need to resolve a merge conflict, need to correct a commit message, forgot to commit a new file). Rebases are difficult to review.

[the remark this answers is not in the document]
Comment (mfil) [reply include/mbedtls/cipher.h:517]: I had to do that a lot, I keep forgetting the signoff...

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM - thanks!

**Review by gilles-peskine-arm — APPROVED** [review]

**Pull request merged**